Fix setState() called after dispose in MacosPulldownButton

Abbas1Hussein opened the pull request:
### Fix: avoid calling setState after dispose in MacosPulldownButton

When I used `MacosPulldownButton`, this error appeared during tests:

```
setState() called after dispose(): _MacosPulldownButtonState (lifecycle state: defunct, not mounted)
```

It happened because `setState` was called inside the `navigator.push(...).then(...)` callback after the widget was already disposed.

I fixed it by adding a mounted check before calling `setState`:

```dart
navigator.push(_pulldownRoute!).then<void>((_) {
  if (!mounted) return;
  setState(() => _pullDownButtonState = PulldownButtonState.enabled);
  _removeMacosPulldownRoute();
});
```

Abbas1Hussein committed:
Fix setState call order in pulldown button

Fix setState() called after dispose in.

Abbas1Hussein committed:
Merge branch 'macosui:dev' into dev

GroovinChip commented:
Hey @Abbas1Hussein, thanks for the PR. This is currently blocked by #580. In the meantime:

- Could you please increment the package version number & update the changelog?
- If you could add a test that covers this change, I'd really appreciate it, so that we can ensure this problem doesn't crop up again

Abbas1Hussein committed:
Update

Fix setState called after dispose in MacosPulldownButton

Abbas1Hussein committed:
Merge branch 'dev' of https://github.com/Abbas1Hussein/macos_ui into dev

Abbas1Hussein commented:
Updated the PR as requested:

- Bumped the version to 2.0.4.
- Updated the CHANGELOG.md.
- Added a test that reproduces the issue and verifies the fix (no more setState after dispose errors).

All tests are passing successfully.

Thanks for the feedback!

Abbas1Hussein committed:
Delete CHANGELOG.md

GroovinChip commented:
Looks like you deleted the changelog, can you please undo that?

Abbas1Hussein committed:
Update CHANGELOG.md

Abbas1Hussein committed:
Merge branch 'dev' of https://github.com/Abbas1Hussein/macos_ui into dev

Abbas1Hussein commented:
I removed the file while cleaning up my branch, I’ve restored the CHANGELOG.md file.

**Please let me know if anything else is needed.**

GroovinChip commented:
> I removed the file while cleaning up my branch, I’ve restored the CHANGELOG.md file.
> 
> **Please let me know if anything else is needed.**

It looks like the file was not restored completely - if you look at the diff on this PR, you will see several recent entries are showing as removed. Please add those back in. Thanks!
